frontend/public/components/replication-controller.jsx

```diff
@@ -27,40 +26,67 @@ import {
 } from './utils';
 
 import { VolumesTable } from './volumes-table';
-import { confirmModal } from './modals';
+import { confirmModal, rollbackRCModal } from './modals';
 import { k8sPatch } from '../module/k8s';
+import { DeploymentConfigModel } from '../models';
 
-const CancelAction = (kind, obj) => ({
-  // t('public~Cancel rollout')
-  labelKey: 'public~Cancel rollout',
-  hidden: !_.includes(
-    ['New', 'Pending', 'Running'],
-    obj?.metadata?.annotations?.['openshift.io/deployment.phase'],
-  ),
-  callback: () =>
-    confirmModal({
-      title: i18next.t('public~Cancel rollout'),
-      message: i18next.t('public~Are you sure you want to cancel this rollout?'),
-      btnText: i18next.t('public~Yes, cancel'),
-      cancelText: i18next.t("public~No, don't cancel"),
-      executeFn: () =>
-        k8sPatch(kind, obj, [
-          {
-            op: 'add',
-            path: '/metadata/annotations/openshift.io~1deployment.cancelled',
-            value: 'true',
-          },
-          {
-            op: 'add',
-            path: '/metadata/annotations/openshift.io~1deployment.status-reason',
-            value: 'cancelled by the user',
-          },
-        ]),
-    }),
-  accessReview: asAccessReview(kind, obj, 'patch'),
-});
+const INACTIVE_STATUSES = ['New', 'Pending', 'Running'];
 
-const menuActions = [CancelAction, ...replicaSetMenuActions];
+const CancelAction = (kind, obj) => {
+  const deploymentPhase = obj?.metadata?.annotations?.['openshift.io/deployment.phase'];
+  return {
+    // t('public~Cancel rollout')
+    labelKey: 'public~Cancel rollout',
+    hidden: !deploymentPhase || !INACTIVE_STATUSES.includes(deploymentPhase),
+    callback: () =>
+      confirmModal({
+        title: i18next.t('public~Cancel rollout'),
+        message: i18next.t('public~Are you sure you want to cancel this rollout?'),
+        btnText: i18next.t('public~Yes, cancel'),
+        cancelText: i18next.t("public~No, don't cancel"),
+        executeFn: () =>
+          k8sPatch(kind, obj, [
+            {
+              op: 'add',
+              path: '/metadata/annotations/openshift.io~1deployment.cancelled',
+              value: 'true',
+            },
+            {
+              op: 'add',
+              path: '/metadata/annotations/openshift.io~1deployment.status-reason',
+              value: 'cancelled by the user',
+            },
+          ]),
+      }),
+    accessReview: asAccessReview(kind, obj, 'patch'),
+  };
+};
+
+const RollbackAction = (kind, obj) => {
+  const deploymentPhase = obj?.metadata?.annotations?.['openshift.io/deployment.phase'];
+  const dcObj = {
```

Comment:
I wouldn't create a fake object... You can specify the access review yourself instead of using the `asAccessReview` helper.

```ts
accessReview: {
  group: DeploymentConfigModel.apiGroup,
  resource: DeploymentConfigModel.plural,
  name: getDeploymentConfigNameForRC(obj),
  namespace: obj?.metadata?.namespace,
}
```